indico/util/reporter.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from copy import deepcopy
+
+from flask import session, request
+
+
+class ReporterBase(object):
+    """Serves as a base class for classes responsible to perform actions on
+    reports
+    """
+
+    #: A unique ID for the report
+    REPORT_ID = ''
+    #: Dict containing the default visible columns and filtering options
+    DEFAULT_REPORT_CONFIG = {}
+    #: OrderedDict containing the filterable columns of the report
+    FILTERABLE_ITEMS = None
+    cache = None
+
+    @classmethod
+    def get_report_config_session_key(cls, entry_parent_id):
+        """Composes the unique configuration ID that will be used as a key to
```

Comment:
We use command tense in docstrings.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

https://www.python.org/dev/peps/pep-0257/